### PR TITLE
Change holidays into datums

### PR DESCRIPTION
**Pull request by MrPerson**
- Holiday datums run a proc called celebrate() when it's time to celebrate them. Currently none of them do anything but that should change, :wink: :wink:
- Holidays can now run for more than a day. The important ones, april fools, christmas, halloween, new years, and easter, all last at least a week. The idea is so people can celebrate christmas in game without having to, you know, actually play on fucking christmas. And also to put a time limit on how long stuff like the annoying spookoween closet skeletons will stick around so it doesn't overstay its welcome and become annoying as shit like last year.
- The event SS now allows more than 1 holiday to run at a time. This matters for new years + christmas, easter + april fools, easter + 4/20, and any holiday that can happen on friday the 13th. The events get stored in a list that's only initialized if there's an active holiday so testing for potential holidays is still pretty easy.
- Added more easter dates so we won't have to add more until 2040. The current batch run out in 2018.

### CODE REVIEW

**Comment by Jalleo** [comment]
:+1: isnt there a equation for when easter happens??

File: code/controllers/subsystem/ticker.dm [hunk 1]
```diff
@@ -47,7 +47,7 @@ var/datum/subsystem/ticker/ticker
 	NEW_SS_GLOBAL(ticker)
 
 	login_music = pickweight(list('sound/ambience/title2.ogg' = 49, 'sound/ambience/title1.ogg' = 49, 'sound/ambience/clown.ogg' = 2)) // choose title music!
-	if(SSevent.holiday == "April Fool's Day")
+	if(SSevent.holidays["April Fool's Day"])
```
Comment (Giacom) [inline code/controllers/subsystem/ticker.dm]: Would be a good time to put these strings into constants.

**Comment by Miauw62** [comment]
>implying spookoween 2014 wasn't the best one in years
Op 25-mrt.-2015 09:59 schreef "Jalleo" <notifications@github.com>:

**Comment by Jordie0608** [comment]
It was the buggiest in years

**Comment by Miauw62** [comment]
that's because it actually had features

**Comment by Jalleo** [comment]
I wasnt implying anything only someone who thinks ahead of the lines do. I liked the buggy spookoween it was funny on multiple levels.

**Comment by MrPerson** [comment]
Yeah but it's all complicated shit written in other languages

**Pull request merged**